lpcompiler.pas

```diff
@@ -2699,7 +2699,7 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
               begin
                 if (Method is TLapeTree_InternalMethod) and
                    TLapeTree_InternalMethod(Method).ForceParam and
-                   (not (Tokenizer.Tok in ReturnOn))
+                   (Tokenizer.Tok <> tk_sym_SemiColon) and (not (Tokenizer.Tok in ReturnOn))
```

Comment:
The additional check shouldn't be necessary here. Semicolon should be part of `ReturnOn`.

```suggestion
                    (not (Tokenizer.Tok in ReturnOn))
```